tests/functional/object/mcg/test_default_backingstore_override.py

```diff
@@ -0,0 +1,179 @@
+        request.config.cache.set("pre_upgrade_alt_bs_name", alt_default_bs_name)
+
+        # 2. Verify the new default is set before the upgrade
+        default_admin_resource = mcg_obj_session.get_admin_default_resource_name()
+        default_bc_bs = mcg_obj_session.get_default_bc_backingstore_name()
+        assert (
+            default_admin_resource == default_bc_bs == alt_default_bs_name
+        ), "The new default backingstore was not overriden before the upgrade!"
+
+    @post_upgrade
+    @polarion_id("OCS-5194")
+    def test_default_backingstore_override_post_upgrade(
+        self,
+        request,
+        mcg_obj_session,
+    ):
+        """
+        Verify the new default is still set post-upgrade
+
+        """
+        # Retrieve the new default backingstore name from the pre-upgrade test
+        alt_default_bs_name = request.config.cache.get("pre_upgrade_alt_bs_name", None)
+
+        # Verify the new default is still set post-upgrade
+        default_admin_resource = mcg_obj_session.get_admin_default_resource_name()
+        default_bc_backingstore = mcg_obj_session.get_default_bc_backingstore_name()
+        assert (
+            default_admin_resource == default_bc_backingstore == alt_default_bs_name
+        ), "The new default backingstore was not preserved after the upgrade!"
+
+    @tier2
+    @skipif_aws_creds_are_missing
+    @polarion_id("OCS-5195")
+    @bugzilla("2237427")
+    def test_bucketclass_replication_after_default_backingstore_override(
+        self,
+        mcg_obj_session,
+        bucket_factory,
+        override_default_backingstore,
+        awscli_pod_session,
+        test_directory_setup,
+    ):
+        """
+        1. Create a target bucket
+        2. Set a bucketclass replication policy to the target bucket on the default bucket class
+        3. Override the default noobaa backingstore
+        4. Create a source OBC under the default bucketclass
+        5. Upload objects to the source bucket and verify they are replicated to the target bucket
+
+        """
+        # 1. Create a target bucket
+        target_bucketclass_dict = {
+            "interface": "OC",
+            "backingstore_dict": {"aws": [(1, "eu-central-1")]},
+        }
+        target_bucket = bucket_factory(bucketclass=target_bucketclass_dict)[0]
+
+        # 2. Set a bucketclass replication policy to the target bucket on the default bucket class
+        replication_policy = {
+            "rules": [
+                {"rule_id": uuid4().hex, "destination_bucket": target_bucket.name}
+            ]
+        }
+        replication_policy_patch_dict = {
+            "spec": {"replicationPolicy": json.dumps(replication_policy)}
+        }
+        OCP(
+            kind=constants.BUCKETCLASS,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.DEFAULT_NOOBAA_BUCKETCLASS,
+        ).patch(params=json.dumps(replication_policy_patch_dict), format_type="merge")
+
```

Comment:
Please use `patch_replication_policy_to_bucketclass` method here which is present in ocs_ci/ocs/bucket_utils.py

Reply:
Thanks for pointing this out! That method was created with an OBC as the target in mind so I made the needed adjustments to use it to patch a bucketclass as well in order to use in `test_bucketclass_replication_after_default_backingstore_override`